applications/crossbar/src/modules/cb_search.erl

```diff
@@ -11,18 +11,19 @@
 -module(cb_search).
 
 -export([init/0
-         ,allowed_methods/0
-         ,resource_exists/0
-         ,validate/1
-         ,authorize/1
+         ,allowed_methods/0 ,allowed_methods/1
+         ,resource_exists/0 ,resource_exists/1
+         ,validate/1 ,validate/2
+         ,authorize/1, authorize/2
         ]).
 
 -include("../crossbar.hrl").
 
 -define(QUERY_TPL, <<"search/search_by_">>).
+-define(MULTI, <<"multi">>).
 
 -define(ACCOUNT_QUERY_OPTIONS, [<<"name">>, <<"number">>, <<"name_and_number">>]).
--define(ACCOUNTS_QUERY_OPTIONS, [<<"name">>]).
+-define(ACCOUNTS_QUERY_OPTIONS, [<<"name">>, <<"realm">>, <<"id">>]).
```

Comment:
These need to be discoverable somehow (same for `ACCOUNT_QUERY_OPTIONS`). Remember, Crossbar is a black box to the clients accessing it; they can't necessarily read the source/docs to figure it out.